packages/gatsby-cypress/README.md

```diff
@@ -3,29 +3,28 @@
 This package provides additional [Cypress](https://cypress.io/) commands for [testing Gatsby websites](/docs/end-to-end-testing/). To use these commands, first install the necessary packages:
 
 ```bash
-npm install cypress gatsby-cypress --save-dev
+npm install cypress gatsby-cypress starter-server-and-test --save-dev
 ```
 
-Once you have installed Cypress, it will automatically create a place to install the Gatsby plugin,
-in `cypress/support/index.js`:
+## Add custom Gatsby testing commands
+
+Next, add a new file located at `cypress/support/index.js` and add the Gatsby-specific Cypress commands:
 
 ```js:title=cypress/support/index.js
 import "gatsby-cypress/commands"
 ```
 
-You also need to expose a `CYPRESS_SUPPORT` environment variable to entirely eliminate any code relating to Cypress in the normal browser build. You can place it in your test script in `package.json`, for example:
-
-```json:title=package.json
-"test": "CYPRESS_SUPPORT=y npm run build && start-server-and-test develop http://localhost:8000 cypress open"
-```
-
 Once imported, the following additional commands are available:
 
 - `cy.getTestElement(selector)`: Selects elements where the `data-testid`
   attribute matches the value of `selector`. Example:
+  
+  ```jsx
+  <button data-testid="btn-to-test">click me</button>
```

Comment:
Nit: but @kentcdodds has expressed some opinions in preferring `getByText` rather than using data-testid. Doesn't need to block this PR though!

Reply:
Yes! Please do avoid test IDs! This one should be selected with `getByText(/click me/i)`

Reply:
Should we remove the `getTestElement` API from `gatsby-cypress`? I was just trying to demonstrate what the exported commands do.

Reply:
Worth keeping it around--there are use cases where it's needed (and we're using them all over the place in our e2e tests 😬), but...

We _should_ create a new API method `getByText` or something similar and expose it and recommend it! However--outside the scope of this PR!

Reply:
We can just use [cypress-testing-library](https://github.com/kentcdodds/cypress-testing-library) like I wrote here 👍 
https://www.gatsbyjs.org/docs/end-to-end-testing/

Reply:
Great - thanks Lennart!

Reply:
So in terms of action steps:

1. Mark this method as not recommended
2. Point to https://github.com/kentcdodds/cypress-testing-library for more powerful commands?

Should I remove the example, too? Or is just the notice enough?

Reply:
I honestly think we should not even mention `getTestElement` in `gatsby-cypress` docs - this is really messy helper that we use in our tests (that probably can be rewritten to use `getByText`).

Only useful thing for users in `gatsby-cypress` really is `cy. waitForRouteChange()`:
In your livestream yesterday you needed to add `cy.wait(500)` when testing `gatsby develop` - `cy.waitForRouteChange()` is more reliable equiavalent of that.

The problem was happening because `cy.visit('/')` resolves when "DOM ready" event is emitted - which most often will happen before react application mounts. `cy.waitForRouteChange()` ensures to wait for react application to mount (on initial visit) and page to actually change on navigation before continuing with tests.

Reply:
@pieh do you have sample code that shows this working? I ended up doing this:

```js
  beforeEach(() => {
    cy.visit(`http://localhost:8000`)
     // Wait for React to finish mounting all components before testing.
    cy.wait(100)
  })
```

This might be specific to `react-helmet`, but when I use `cy.waitForRouteChange()` the Helmet stuff hasn't happened yet. If I use `cy.wait(100)`, it has.

Reply:
It might be specific to `react-helmet`, it does use `requestAnimationFrame` to update `head`/DOM under the hood. For the most part I was referring to part when `cypress-axe` didn't pick anything up in `gatsby develop` (no landmarks rules, etc).

We do use `waitForRouteChange` for all of our e2e tests (for example in https://github.com/gatsbyjs/gatsby/blob/a4f03cfe0d76a6feb7be76ed155a16a225827941/e2e-tests/development-runtime/cypress/integration/navigation/linking.js#L2-L4 ) for both development and production runtime tests and it works there (and remove flakiness of those tests - we were using `cy.wait` but it would intermittently fail in cases when wait time was not enough in CircleCI). But we don't have any tests that would check anything related to `react-helmet` (as it's not part of core `gatsby` package), so probably that's why we didn't hit the issue.